addons/mail/models/mail_message.py

```diff
@@ -1100,6 +1100,7 @@ def message_format(self):
                 'is_discussion': message_sudo.subtype_id.id == com_id,
                 'subtype_description': message_sudo.subtype_id.description,
                 'is_notification': vals['message_type'] == 'user_notification',
+                'channel_types': list(message_sudo.mapped('channel_ids.channel_type')),
```

Comment:
Why do you have to specify "list()" here?

Reply:
Also, I don't know discuss in details but I'm not sure why this information has to be in every single messages.
Surely when you display the message there is a way to know in which channel and channel type you are to condition some behavior right?

Seems like it adds complexity / potential overhead where it doesn't need to.

Reply:
@awa-odoo 
In JS, in mail message, I only have access to the ID (int) of the channel, so if I want to know if the message is inside a social channel, I'll need to do a RPC call :/

I use this information to show the `anonymous_name` of the channel instead of the name of the partner ("public partner") if the message is inside a social channel.
(In livechat the situation is not the same because the name of the channel is the name of the partner...-

The other solution is to inherit from `'mail.message' -> message_format` in social and to add the field (like I did before, but people didn't like it).

The last solution would be to "include" `MailManager` , and to change the method `_makeMessage` to return a new message type (with the social information) if we are in a social channel. But, I'll need to create one more JS class (and it will probably be broken after the discuss refactoring).

Reply:
(sorry for the very long message :/ )